### PR TITLE
Fix Keras Import for Custom Loss Gradient

### LINKED ISSUE #809: Missing Keras import guard in KerasClassifier's custom_loss_gradient function.
Hello! While we were testing one of the ART poisoning attacks (Feature Collision) we encountered an import issue with the `self.estimator.custom_loss_gradient` function:

Line 633 of `keras.py` (master branch):
 `        import keras.backend as k`

We're currently using TensorFlow v2 which uses `import tf.keras` commands instead. 
We fixed the issue locally by swapping line 633 with the following lines :

```
        if self.is_tensorflow:
            import tensorflow.keras.backend as k
        else:
            import keras.backend as k
```

Let me know if this correction can be applied into `keras.py`, thanks! 
We're currently using ART release 1.5. 
Let me know if this change can be applied into the 1.5.x version, or if it will apply in release 1.6.

Thanks,
Howard

### PR DESCRIPTION
**Pull request by ebubae**
# Description

This PR fixes an issue with inputs when using custom loss gradients.

Fixes #809 

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — APPROVED** [review]
Hi @ebubae Thank you very much, this looks good to me.
